src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -299,28 +266,13 @@ func onBootstrapControlPlane(s *state.State, initConfig map[string]string) error
 	}
 
 	// Start services
-	switch cfg.APIServer.Datastore {
-	case "k8s-dqlite":
-		if err := snaputil.StartK8sDqliteServices(s.Context, snap); err != nil {
-			return fmt.Errorf("failed to start control plane services: %w", err)
-		}
-	case "external":
-	default:
-		return fmt.Errorf("unsupported datastore %s, must be one of %v", cfg.APIServer.Datastore, setup.SupportedDatastores)
-	}
-
-	if err := snaputil.StartControlPlaneServices(s.Context, snap); err != nil {
-		return fmt.Errorf("failed to start control plane services: %w", err)
-	}
-
-	// Wait for API server to come up
-	client, err := k8s.NewClient(snap)
-	if err != nil {
-		return fmt.Errorf("failed to create k8s client: %w", err)
+	if err := startControlPlaneServices(s.Context, snap, cfg.APIServer.Datastore); err != nil {
+		return fmt.Errorf("failed to start services: %w", err)
 	}
 
-	if err := client.WaitApiServerReady(s.Context); err != nil {
-		return fmt.Errorf("k8s api server did not become ready in time: %w", err)
+	// Wait until Kube-API server is ready
+	if err := waitReadyApiServer(s.Context, snap); err != nil {
+		return fmt.Errorf("failed to wait for Kube-API server: %w", err)
```

Comment:
Let's be consistent with `kube-apiserver` in the error messages, using different similar names in different places causes confusion.

```suggestion
	if err := waitReadyApiServer(s.Context, snap); err != nil {
		return fmt.Errorf("failed to wait for kube-apiserver: %w", err)
```